### PR TITLE
[CI] [GHA] Introduce GHA Linux nGraph ONNX Pipeline

### PR DESCRIPTION
**Pull request by akashchi**
### Details:
This PR introduces the Linux nGraph ONNX pipeline in the GitHub Actions environment. It is mirroring the [Azure Linux nGraph ONNX pipeline](https://github.com/openvinotoolkit/openvino/blob/master/.ci/azure/linux_ngraph_onnx.yml).

### Tickets:
 - 117534

### CODE REVIEW

**Review by mryzhov — APPROVED** [review]
Could we execute build and build docker in parallel? And test step depends on them?

File: .github/workflows/linux_ngraph_onnx.yml [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        with:
+          path: 'openvino'
+          submodules: 'recursive'
+
+      - name: Create Directories
+        run: |
+          mkdir -p ${{ env.BUILD_DIR }}
+          mkdir -p ${{ env.INSTALL_DIR }}
+          mkdir -p ${{ env.MODELS_PATH }}
+          mkdir -p ${{ env.MODELS_PATH }}/models_data
```
Comment (ilya-lavrenov) [inline .github/workflows/linux_ngraph_onnx.yml:69]: I suppose it's not required

File: .github/workflows/linux_ngraph_onnx.yml [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    runs-on: ubuntu-20.04-8-cores
+    env:
+      CMAKE_BUILD_TYPE: 'Release'
+      CMAKE_GENERATOR: 'Ninja'
+      OPENVINO_REPO: ${{ github.workspace }}/openvino
+      BUILD_DIR: ${{ github.workspace }}/build
+      INSTALL_DIR: ${{ github.workspace }}/install
+      OV_TEMP: ${{ github.workspace }}/openvino_temp
+      DATA_PATH: ${{ github.workspace }}/testdata
+      MODELS_PATH: ${{ github.workspace }}/testdata
```
Comment (ilya-lavrenov) [inline .github/workflows/linux_ngraph_onnx.yml:53]: some of variables are not used

File: .github/workflows/linux_ngraph_onnx.yml [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+                    
+          wget https://github.com/ninja-build/ninja/releases/download/v1.10.2/ninja-linux.zip
+          unzip ninja-linux.zip
+          sudo cp -v ninja /usr/local/bin/
+
+      - name: Get tools versions
+        run: |
+          ninja --version
+          python3 --version
+          cmake --version
```
Comment (ilya-lavrenov) [inline .github/workflows/linux_ngraph_onnx.yml:98]: remove

File: .github/workflows/linux_ngraph_onnx.yml [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+        run: ${{ env.OPENVINO_REPO }}/src/frontends/onnx/tests/tests_python/model_zoo_preprocess.sh -d ${{ env.MODELS_PATH }}/models_data -o -s "${{ env.ONNX_MODEL_ZOO_SHA }}"
+
+      - name: Get number of CPU cores
+        uses: SimenB/github-actions-cpu-cores@v1
+        id: cpu-cores
+
+      # Docker build
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
```
Comment (ilya-lavrenov) [inline .github/workflows/linux_ngraph_onnx.yml:115]: why do we need qemu?

File: .github/workflows/linux_ngraph_onnx.yml [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+      - name: Check directories
+        run: |
+          ls -laR ${{ env.MODELS_PATH }}
+
+      - name: Run tests via tox
+        run: |
+          sudo docker run \
+            --name openvino-onnx-ci-container \
+            --volume ${{ env.MODELS_PATH }}/models_data/model_zoo/onnx_model_zoo_${{ env.ONNX_MODEL_ZOO_SHA }}:/root/.onnx/model_zoo/onnx_model_zoo \
+            --volume ${{ env.MODELS_PATH }}/msft:/root/.onnx/model_zoo/MSFT openvino-onnx-ci-image:latest \
```
Comment (ilya-lavrenov) [inline .github/workflows/linux_ngraph_onnx.yml:140]: honestly, I don't understand why do we need docker in the original pipeline..
maybe we need to simplify the workflow and keep as a job in linux.yml

**Pull request closed without merging**